src/modules/ods/ods.controller.test.ts

```diff
@@ -0,0 +1,30 @@
+import { CUSTOMERS } from '@ukef/constants';
+import { when } from 'jest-when';
+import { OdsController } from './ods.controller';
+import { OdsService } from './ods.service';
+
+describe('OdsController', () => {
+  let odsServiceFindCustomer: jest.Mock;
+
+  let controller: OdsController;
+
+  beforeEach(() => {
+    const odsService = new OdsService(null, null);
+    odsServiceFindCustomer = jest.fn();
+    odsService.findCustomer = odsServiceFindCustomer;
+
+    controller = new OdsController(odsService);
+  });
+
+  describe('findCustomer', () => {
+    it('should return the customer when a valid customer URN is provided', async () => {
```

Comment:
Can we add an assertion with invalid and short URNs i.e. few unhappy paths please.

Reply:
all of these will be covered by the API tests as these are testing the decorators on the input and we don't do that anywhere else in mdm-api in the unit tests (it would require me setting up the validation pipeline within the unit test which would be a new pattern)